### DISCUSSION TITLE
download_dsyms doesn't wait for dSYM files to process

### DISCUSSION THREAD

**Discussion by jonasz88** [fastlane/fastlane#20263] [Q&A]
Hi,

I don't know if this is a bug or something wrong with my code / environment, so I decided to write about this problem here first. 

I have the following lane which aims to download the dSYM files from App Store Connect, and then upload them to Crashlytics. (Because my app has Bitcode enabled.)

```
lane :upload_symbols do
   
    version = get_version_number()

    download_dsyms(
        api_key: <my API key>,
        version: version, 
	build_number: <build number>,
	app_identifier: <app identifier>,
	wait_for_dsym_processing: true,
	wait_timeout: 300
     )

     upload_symbols_to_crashlytics(
         gsp_path: <path to .plist file in my project>
     )

     clean_build_artifacts() 
end

```

If I understand correctly, setting the `wait_for_dsym_processing` flag to `true` should make my lane wait for the dSYM files to be ready to use and then download them. 

When I run it on my CI (GitLab CI) or even locally, on my MacBook Pro, I see this in the logs:

![1](https://user-images.githubusercontent.com/7733420/167698546-8f6527a2-31bf-4cb5-8976-8a45aed8d86c.png)

It looks like Fastlane immediately skips to the next action and uploads the dSYM files, but these earlier generated by Xcode. If I run this lane for the second time, after the processing is completed on Apple's side and the files are available for this app version, the dSYMs are being downloaded correctly and looks like everything is okay then.

The Fastlane version I use is 2.205.2 and `fastlane-plugin-firebase_app_distribution` is 0.3.4.

Has anyone run into the same problem? 

Tomek

**Comment by SzymonGajdzica** [comment]
Got the same problem

**Comment by shermanchow1999** [comment]
The value of version should be set to 'latest' or 'live'.

**Reply by jonasz88** [reply]
Could you please elaborate on that a little more? I guess that it won't resolve my problem, because I don't need neither the dSYMs for the latest build nor the live one. 

The problem here is that, for some reason, Fastlane cannot download dSYMs for builds that are being processed right after uploading on App Store Connect (TestFlight) 😟.

**Comment by coadtan** [comment]
I'm having the same issue. Have you found any workaround or fix.

Thank you.
